rotkehlchen/chain/bitcoin/utils.py

```diff
@@ -264,3 +285,127 @@ def scriptpubkey_to_btc_address(data: bytes) -> BTCAddress:
         return scriptpubkey_to_p2sh_address(data)
 
     return scriptpubkey_to_bech32_address(data)
+
+
+def query_apis_via_callbacks(api_callbacks: dict[str, Callable], *args, **kwargs):
+    errors: dict[str, str] = {}
+    for api_name, callback in api_callbacks.items():
+        try:
+            return callback(*args, **kwargs)
+        except (
+            requests.exceptions.RequestException,
+            UnableToDecryptRemoteData,
+            requests.exceptions.Timeout,
+            RemoteError,
+            DeserializationError,
+        ) as e:
+            errors[api_name] = str(e)
+            continue
+        except KeyError as e:
+            errors[api_name] = f"Got unexpected response from {api_name}. Couldn't find key {e!s}"
+
+    serialized_errors = ', '.join(f'{source} error is: "{error}"' for (source, error) in errors.items())  # noqa: E501
+    raise RemoteError(f'Bitcoin external API request for balances failed. {serialized_errors}')
+
+
+def query_blockstream_or_mempool_api(url_suffix: str) -> dict | list:
```

Comment:
since balances & transactions will be handled by `BitcoinManager` class, i don't think this should be in utils.py anymore.